WordPress/src/test/java/org/wordpress/android/ui/jetpack/scan/builders/ScanStateListItemsBuilderTest.kt

```diff
@@ -146,43 +157,324 @@ class ScanStateListItemsBuilderTest : BaseUnitTest() {
+            HeaderState(UiStringRes(R.string.scan_idle_threats_found_title))
         )
+    }
 
-        assertThat(scanStateItems.filterIsInstance(ThreatItemState::class.java).first().subHeader)
-            .isEqualTo(subHeader.text)
+    @Test
+    fun `given idle state with threats, when items are built, items contain threats found description`() = test {
+        buildScanStateItems(scanStateModelWithThreats)
+
+        verify(htmlMessageUtils).getHtmlMessageFromStringFormatResId(
+            R.string.scan_idle_with_threats_description,
+            "<b>${threats.size}</b>",
+            "<b>${site.name ?: resourceProvider.getString(R.string.scan_this_site)}</b>",
+            resourceProvider.getString(R.string.scan_here_to_help)
+        )
     }
 
-    /* IDLE - THREATS FOUND STATE */
+    @Test
+    fun `given idle state with threats, when items are built, items contain scan now button`() = test {
+        val scanStateItems = buildScanStateItems(scanStateModelWithThreats)
+
+        assertThat(scanStateItems.filterIsInstance(ActionButtonState::class.java).map { it.text }).contains(
+            UiStringRes(R.string.scan_again)
+        )
+    }
 
     @Test
-    fun `builds clickable text info in description for scan state model with threats`() = test {
-        val clickableText = "clickable text"
-        val descriptionWithClickableText = "description with $clickableText"
-        whenever(htmlMessageUtils.getHtmlMessageFromStringFormatResId(anyInt(), any()))
-            .thenReturn(descriptionWithClickableText)
-        whenever(resourceProvider.getString(R.string.scan_here_to_help)).thenReturn(clickableText)
+    fun `given idle state with fixable threats, when items are built, items contain fix all threats button`() = test {
+        val threats = listOf(threat.copy(baseThreatModel = baseThreatModel.copy(fixable = mock())))
+        val scanStateModelWithFixableThreats = scanStateModelWithThreats.copy(threats = threats)
 
-        val scanStateItems = buildScanStateItems(scanStateModelWithThreats)
+        val scanStateItems = buildScanStateItems(scanStateModelWithFixableThreats)
+
+        assertThat(scanStateItems.filterIsInstance(ActionButtonState::class.java).map { it.text }).contains(
+            UiStringRes(R.string.threats_fix_all)
+        )
+    }
+
+    @Test
+    fun `given idle state with not fixable threats, when items are built, items contain fix all threats button`() =
+        test {
+            val threats = listOf(threat.copy(baseThreatModel = baseThreatModel.copy(fixable = null)))
+            val scanStateModelWithNoFixableThreats = scanStateModelWithThreats.copy(threats = threats)
+
+            val scanStateItems = buildScanStateItems(scanStateModelWithNoFixableThreats)
+
+            assertThat(
+                scanStateItems.filterIsInstance(ActionButtonState::class.java)
+                    .map { it.text }
+            ).doesNotContain(
```

Comment:
Minor (🔍): Consider fitting this in one line, just like you did with the above test. I am seeing it to actually fit in one line, isn't it?

Reply:
Fitted in 37ac2d4